packages/core/src/agent/DextoAgent.ts

```diff
@@ -701,10 +959,93 @@ export class DextoAgent {
         return await this.sessionManager.getSessionTitle(sessionId);
     }
 
+    /**
+     * Generate a title for a session on-demand.
+     * Uses the first user message content to generate a descriptive title.
+     *
+     * @param sessionId Session ID to generate title for
+     * @returns Promise that resolves to the generated title, or null if generation failed
+     */
+    public async generateSessionTitle(sessionId: string): Promise<string | null> {
```

Comment:
we had some maybegeneratetitle method. what happened to that? does it need to be deleted